pymc4/variational/approximations.py

```diff
@@ -42,25 +41,38 @@ def __init__(self, model: Optional[Model] = None, random_seed: Optional[int] = N
 
         self.order = ArrayOrdering(self.state.all_unobserved_values)
         self.unobserved_keys = self.state.all_unobserved_values.keys()
-        self.target_log_prob = self._build_logfn()
+        self.target_log_prob, self.deterministics_callback = self._build_logp_and_deterministic_fn()
         self.approx = self._build_posterior()
 
-    def _build_logfn(self):
-        """Build vectorized logp function."""
+    def _build_logp_and_deterministic_fn(self):
+        """Build vectorized logp and deterministic functions."""
 
         @tf.function(autograph=False)
         def logpfn(*values):
             split_view = self.order.split(values[0])
             _, st = flow.evaluate_meta_model(self.model, values=split_view)
             return st.collect_log_prob()
 
-        def vectorize_logp_function(logpfn):
-            def vectorized_logpfn(*q_samples):
-                return tf.vectorized_map(lambda samples: logpfn(*samples), q_samples)
+        @tf.function(autograph=False)
```

Comment:
Let's also add a test to make sure it works

Reply:
Resolved in 6189ba1.